src/imap/model.rs

```diff
@@ -50,14 +70,41 @@ impl<'a> ImapConnector<'a> {
         Ok(Self { account, sess })
     }
 
+    /// Closes the connection.
+    ///
+    /// **ALWAYS CALL THIS IF YOU DON'T NEED THE CONNECTION ANYMORE!**
     pub fn logout(&mut self) {
         debug!("logout");
         match self.sess.logout() {
             _ => (),
         }
     }
 
-    pub fn set_flags(&mut self, mbox: &str, uid_seq: &str, flags: &str) -> Result<()> {
+    /// Applies the given flags to the msg.
+    ///
+    /// # Example
+    /// ```no_run
+    /// use himalaya::imap::model::ImapConnector;
+    /// use himalaya::config::model::Account;
+    /// use himalaya::flag::model::Flags;
+    /// use imap::types::Flag;
+    ///
+    /// fn main() {
+    ///     let account = Account::default();
+    ///     let mut imap_conn = ImapConnector::new(&account).unwrap();
+    ///     let flags = Flags::from(vec![Flag::Seen]);
+    ///
+    ///     // Mark the message with the UID 42 in the mailbox "rofl" as "Seen" and wipe all other
+    ///     // flags
+    ///     imap_conn.set_flags("rofl", "42", flags).unwrap();
+    ///
+    ///     imap_conn.logout();
+    /// }
+    /// ```
+    pub fn set_flags(&mut self, mbox: &str, uid_seq: &str, flags: Flags) -> Result<()> {
+
```

Comment:
Why adding a new line here? It does not really fit the global style of the code base.